core/Plugin/ControllerAdmin.php

```diff
@@ -213,6 +215,43 @@ private static function isUsingPhpVersionCompatibleWithNextPiwik()
         return version_compare( PHP_VERSION, self::getNextRequiredMinimumPHP(), '>=' );
     }
 
+    public static function notifyPiwik2EndOfLife()
+    {
+        if (!Piwik::isUserHasSomeViewAccess()) {
+            return;
+        }
```

Comment:
Should this show for anonymous users?

Reply:
I would say so. I think we discussed it in the chat but not 100% sure. Maybe @mattab can confirm.